notes/2018-01-22-packaging.md

```diff
@@ -0,0 +1,104 @@
+Packaging
+=========
+
+This documents explains how we want to package and distribute Radicle.
+
+We will create packages for Brew, Debian, Pacman, and (possibly Nix). This
+packages will be made available through Github Releases.
+
+Haskell binaries will be statically linked with `-O3` optimization.
```

Comment:
I didn't know `-O3` actually existed, and it is usually advised to use `-O2` over `-O1` **iff** you have **strong empirical evidence** that it doesn't make things worse. So, do we have empirical evidence?

Reply:
Oh and, do we have evidence that all our binaries _can_ be statically linked?

Reply:
Yeah, I always used `-O2` myself as well. Interestingly enough, `-O3` is actually not even mentioned in the last [GHC user guide](https://downloads.haskell.org/~ghc/master/users-guide/using-optimisation.html#o-convenient-packages-of-optimisation-flags). Perhaps is more conservative to go with `-O2`? I don't think it would make such a drastic difference anyway.

Reply:
I changed the optimization to use `-O2`. Build time should not be too much of a concern since this still runs in only a few minutes on CI.

> Oh and, do we have evidence that all our binaries _can_ be statically linked?

Yes, everything that we want to distribute can be statically linked. (We will not distribute the old server. It uses libpq which cannot not be statically linked)

Reply:
My question still persists ;)

Reply:
> My question still persists ;)

Which question are you referring to? If it is about empirical evidence for `-O2` being better than `-O1`, then no, we don’t. If it is about evidence that the binaries can be statically linked, then I’d say yes. I built them using static linking and was able to run them on at two distributions